src/format/KeePass2Repair.cpp

```diff
@@ -29,8 +29,14 @@ KeePass2Repair::KeePass2Repair()
 {
 }
 
+KeePass2Repair::~KeePass2Repair()
+{
+    delete m_db;
+}
+
 KeePass2Repair::RepairResult KeePass2Repair::repairDatabase(QIODevice* device, const CompositeKey& key)
 {
+    delete m_db;
```

Comment:
Call
```
m_db.reset(reader.readDatabase(device, key, true));
```
instead. Replace further deletions of `m_db` within this file with `reset()` as well.

Reply:
done, but kept temporary `db` object not to store the database after repair failed